cmd/limits.go

```diff
@@ -0,0 +1,117 @@
+package cmd
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	k8sResource "k8s.io/apimachinery/pkg/api/resource"
+)
+
+type limitFlags struct {
+	cpuArg    string
+	cpu       k8sResource.Quantity
+	memoryArg string
+	memory    k8sResource.Quantity
+}
+
+var limitConfig limitFlags
+
+func (limit *limitFlags) parseLimitFlags() {
+	if len(limit.cpuArg) != 0 {
+		quantity, err := k8sResource.ParseQuantity(limit.cpuArg)
+		if err != nil {
+			log.Error("Wrong cpu argument: " + err.Error())
+			return
+		}
+		limit.cpu = quantity
+	}
+
+	if len(limit.memoryArg) != 0 {
+		quantity, err := k8sResource.ParseQuantity(limit.memoryArg)
+		if err != nil {
+			log.Error("Wrong cpu argument: " + err.Error())
+			return
+		}
+		limit.memory = quantity
+	}
+}
+
+func checkLimits(container Container, limits limitFlags, result *Result) {
+	if container.Resources.Limits == nil {
+		occ := Occurrence{id: ErrorResourcesLimitsNIL, kind: Warn, message: "Resource limit not set, please set it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+		return
+	}
+
+	checkCPULimit(container, limits, result)
+	checkMemoryLimit(container, limits, result)
+}
+
+func checkCPULimit(container Container, limits limitFlags, result *Result) {
+	cpu := container.Resources.Limits.Cpu()
+	if cpu == nil || cpu.IsZero() {
+		occ := Occurrence{id: ErrorResourcesLimitsCpuNIL, kind: Warn, message: "CPU limit not set, please set it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+		return
+	}
+
+	if limits.cpu.MilliValue() > 0 && cpu.MilliValue() > limits.cpu.MilliValue() {
+		message := fmt.Sprintf("CPU limit exceeded, it is set to %s but it must not exceed %s. Please adjust it!", cpu.String(), limits.cpu.String())
+		occ := Occurrence{id: ErrorResourcesLimitsCpuExceeded, kind: Warn, message: message}
+		result.Occurrences = append(result.Occurrences, occ)
```

Comment:
If you want to make it easy for people to parse the violations could log them in the result object and them print them as for example CPULimit, CPUActual or something like this take a look at this: https://github.com/Shopify/kubeaudit/blob/master/cmd/util.go#L119-L148

Reply:
I didn't see that. It's better to process the results.